xonsh/tools.py

```diff
@@ -280,6 +280,7 @@ def add(self, data, front=False, replace=False):
         None
 
         """
+        data = str(expand_path(data))
```

Comment:
Added this to account for the pathlib stuff so its always expanded and converted to a string before the comparison